### ISSUE TITLE
Error when trying to reload a default style or metadata from GeoNode

### ISSUE THREAD

**Issue by Gpetrak** [GeoNode/QGISGeoNodePlugin#285]
Although the style upload in GeoNode works, when the user tries to get the default style by selecting the button: “Reload the default style from GeoNode” the plugin raises the following error:

```bash
Traceback (most recent call last):
  File "/home/gpetr/.local/share/QGIS/QGIS3/profiles/test/python/plugins/qgis_geonode/gui/geonode_map_layer_config_widget.py", line 171, in handle_style_downloaded
    self.apply()
  File "/home/gpetr/.local/share/QGIS/QGIS3/profiles/test/python/plugins/qgis_geonode/gui/geonode_map_layer_config_widget.py", line 127, in apply
    self._apply_sld()
  File "/home/gpetr/.local/share/QGIS/QGIS3/profiles/test/python/plugins/qgis_geonode/gui/geonode_map_layer_config_widget.py", line 437, in _apply_sld
    layer_properties_dialog.syncToLayer()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'QFrame' object has no attribute 'syncToLayer'
```

Also, when the user tries to reload metadata by selecting the button: “Reload metadata from GeoNode” it gets a quite similar error:

```bash
Traceback (most recent call last):
  File "/home/gpetr/.local/share/QGIS/QGIS3/profiles/test/python/plugins/qgis_geonode/gui/geonode_map_layer_config_widget.py", line 311, in handle_metadata_downloaded
    self.apply()
  File "/home/gpetr/.local/share/QGIS/QGIS3/profiles/test/python/plugins/qgis_geonode/gui/geonode_map_layer_config_widget.py", line 127, in apply
    self._apply_sld()
  File "/home/gpetr/.local/share/QGIS/QGIS3/profiles/test/python/plugins/qgis_geonode/gui/geonode_map_layer_config_widget.py", line 432, in _apply_sld
    sld_load_result = self.layer.readSld(
                      ^^^^^^^^^^^^^^^^^^^
TypeError: readSld(self, node: QDomNode, errorMessage: str): argument 1 has unexpected type 'NoneType'
```

**Comment by giohappy** [comment]
does it happen always or erratically @Gpetrak

**Comment by Gpetrak** [comment]
@giohappy this happens always.

**Comment by giohappy** [comment]
@Gpetrak I suspect the first problem is related to this hacky code:

https://github.com/GeoNode/QGISGeoNodePlugin/blob/6eed6e3d1a1e066657997767c242d052530f985c/src/qgis_geonode/gui/geonode_map_layer_config_widget.py#L452-L455

Maybe this traversal doesn't work for other reasons here (an error stream with the style retrieval, or whatever). In any case this solution is very fragile. A more robust traversal should be implemented, maybe an iterative loop where the type of the parent class is checked to obtain the one searched for.

**Comment by Gpetrak** [comment]
Thank you @giohappy. Finally, I fixed the error of `SLD` and `Metadata` and I implemented a more elegant way in order to find the correct type of object, following your recommendation.
The corresponding PR of this issue is available here #294
